appinventor/appengine/src/com/google/appinventor/client/youngandroid/TextValidators.java

```diff
@@ -23,6 +23,7 @@
 
   private static final int MAX_FILENAME_SIZE = 100;
   private static final int MIN_FILENAME_SIZE = 1;
+  public static String INVALID_PROJECT_NAME_TYPE;
```

Comment:
Rather than creating a public static field to hold an error, I would strongly recommend we move to using an `enum` to capture the various error conditions (with one `SUCCESS = 0` entry). Then, the error messages should be internationalized.